classes/converter.php

```diff
@@ -550,13 +471,23 @@ public function set_usesdkcreds($value) {
     public function define_client_check_sdk() {
         global $OUTPUT;
         $text = '';
-        $converter = new \fileconverter_librelambda\converter();
-        if (!$converter->get_usesdkcreds()) {
-            if ($converter::are_requirements_met('SDK')) {
+        if (!$this->get_usesdkcreds()) {
+            $this->remove_client();
+            $this->set_usesdkcreds(true);
+            // First check that we have the basic configuration settings set.
+            if (!$this->is_config_set()) {
```

Comment:
There can be a scenario when the config is not set, but we set `usesdkcreds` to true and return. It might bite us later as we didn't set `usesdkcreds` back to false.
It looks like this check should be called before `$this->remove_client();`. If config is set then we can remove client, set `usesdkcreds` to true, test, get `usesdkcreds` back to false and return.

Reply:
Oh yeah. That's how it bites:

![image](https://user-images.githubusercontent.com/22066290/96095903-f80b7980-0f1a-11eb-9150-86e521ac4e15.png)

When config is not set api_key and api_secret fields are hidden